x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/QueryTranslator.java

```diff
@@ -65,11 +68,27 @@ public static Query toQuery(Expression e) {
 
     public static Query toQuery(Expression e, TranslatorHandler handler) {
         Query translation = null;
-        for (ExpressionTranslator<?> translator : QUERY_TRANSLATORS) {
-            translation = translator.translate(e, handler);
-            if (translation != null) {
-                return translation;
+        int i = 0;
+        while (translation == null && i < QUERY_TRANSLATORS.size()) {
+            translation = QUERY_TRANSLATORS.get(i).translate(e, handler);
+            i++;
+        }
```

Comment:
Why move from the iterable to the index based approach? I see the later as a downgrade.